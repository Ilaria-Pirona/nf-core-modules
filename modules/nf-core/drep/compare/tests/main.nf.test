// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test drep/compare
nextflow_process {

    name "Test Process DREP_COMPARE"
    script "../main.nf"
    process "DREP_COMPARE"

    tag "modules"
    tag "modules_nfcore"
    tag "drep"
    tag "drep/compare"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("drep compare test - fasta") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [file(params.modules_testdata_base_path + 'genomics/prokaryotes/haemophilus_influenzae/genome/genome.fna.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)]

                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("drep compare test - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    //file(params.modules_testdata_base_path + 'genomics/prokaryotes/haemophilus_influenzae/genome/genome.fna.gz', checkIfExists: true),
                    //file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                                        file('/Users/ilariapirona/Desktop/nf-core-modules/modules/nf-core/drep/compare/tests/10037_SAMEA2241279.fa', checkIfExists: true),
                    file('/Users/ilariapirona/Desktop/nf-core-modules/modules/nf-core/drep/compare/tests/10036_SAMEA2241278.fa', checkIfExists: true),
                    file('/Users/ilariapirona/Desktop/nf-core-modules/modules/nf-core/drep/compare/tests/10035_SAMEA2241277.fa', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
