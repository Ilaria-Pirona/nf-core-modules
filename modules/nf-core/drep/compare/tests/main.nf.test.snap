{
    "drep compare test - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "data_table": [
                    
                ],
                "figures": [
                    
                ],
                "log_file": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T10:45:50.28061"
    },
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "data_table": [
                    
                ],
                "figures": [
                    
                ],
                "log_file": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T10:38:26.172936"
    },
    "drep compare test - fasta": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "data_table": [
                    
                ],
                "figures": [
                    
                ],
                "log_file": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T10:45:36.826394"
    },
    "sarscov2 - bam": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "data_table": [
                    
                ],
                "figures": [
                    
                ],
                "log_file": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-29T10:38:14.584367"
    }
}